libraries/USB/src/USBMIDI.cpp

```diff
@@ -0,0 +1,140 @@
+                               _(velocity)};
+    writePacket(&event);
+}
+
+// Program Change
+void USBMIDI::programChange(uint8_t program, uint8_t channel) {
+    midiEventPacket_t event = {MIDI_CIN_PROGRAM_CHANGE, STATUS(MIDI_CIN_PROGRAM_CHANGE, channel),
+                               _(program), 0x0};
+    writePacket(&event);
+}
+
+// Control Change (Continuous Controller)
+void USBMIDI::controlChange(uint8_t control, uint8_t value, uint8_t channel) {
+    midiEventPacket_t event = {MIDI_CIN_CONTROL_CHANGE, STATUS(MIDI_CIN_CONTROL_CHANGE, channel),
+                               _(control), _(value)};
+    writePacket(&event);
+}
+
+// Polyphonic Key Pressure (Aftertouch)
+void USBMIDI::polyPressure(uint8_t note, uint8_t pressure, uint8_t channel) {
+    midiEventPacket_t event = {MIDI_CIN_POLY_KEYPRESS, STATUS(MIDI_CIN_POLY_KEYPRESS, channel), _(note),
+                               _(pressure)};
+    writePacket(&event);
+}
+
+// Channel Pressure (Aftertouch)
+void USBMIDI::channelPressure(uint8_t pressure, uint8_t channel) {
+    midiEventPacket_t event = {MIDI_CIN_CHANNEL_PRESSURE, STATUS(MIDI_CIN_CHANNEL_PRESSURE, channel),
+                               _(pressure), 0x0};
+    writePacket(&event);
+}
+
+// Pitch Bend Change [-8192,0,8191]
+void USBMIDI::pitchBend(int16_t value, uint8_t channel) {
+    uint16_t pitchBendValue = constrain(value, -8192, 8191) + 8192;
+    pitchBend(pitchBendValue);
+}
+
+// Pitch Bend Change [0,8192,16383]
+void USBMIDI::pitchBend(uint16_t value, uint8_t channel) {
+    uint16_t pitchBendValue = static_cast<uint16_t>(uconstrain(value, 0, 16383));
+    // Split the 14-bit integer into two 7-bit values
+    uint8_t lsb = pitchBendValue & 0x7F;        // Lower 7 bits
+    uint8_t msb = (pitchBendValue >> 7) & 0x7F; // Upper 7 bits
+
+    midiEventPacket_t event = {MIDI_CIN_PITCH_BEND_CHANGE, STATUS(MIDI_CIN_PITCH_BEND_CHANGE, channel),
+                               lsb, msb};
+    writePacket(&event);
+}
+
+// Pitch Bend Change [-1.0,0,1.0]
+void USBMIDI::pitchBend(double value, uint8_t channel) {
+    // Multiply by 8191 and round to nearest integer
+    int16_t pitchBendValue = static_cast<int16_t>(round(constrain(value, -1.0, 1.0) * 8191.0));
+
+    pitchBend(pitchBendValue, channel);
+}
+
+bool USBMIDI::readPacket(midiEventPacket_t *packet) {
+    return tud_midi_packet_read((uint8_t *)packet);
+}
+
+bool USBMIDI::writePacket(midiEventPacket_t *packet) {
+    return tud_midi_packet_write((uint8_t *)packet);
+}
+
+// Default Cable Number (for simplified APIs that do not expose this)
+#define DEFAULT_CN 0
+
+size_t USBMIDI::write(uint8_t c) {
+    // MIDI_CIN_1BYTE_DATA => Verbatim MIDI byte-stream copy
+    // (See also Table 4-1 of USB MIDI spec 1.0)
+    midiEventPacket_t packet = {DEFAULT_CN | MIDI_CIN_1BYTE_DATA, c, 0, 0};
+
+    return tud_midi_packet_write((uint8_t *)&packet);
+}
+
+#endif // CONFIG_TINYUSB_MIDI_ENABLED
```

Comment:
Close the #if from above comment
```suggestion
#endif /* CONFIG_TINYUSB_MIDI_ENABLED */
#endif /* SOC_USB_OTG_SUPPORTED */
```